src/Microsoft.AspNet.Server.Kestrel/Filter/StreamSocketOutput.cs

```diff
@@ -62,6 +95,7 @@ public MemoryPoolIterator2 ProducingStart()
 
         public void ProducingComplete(MemoryPoolIterator2 end)
         {
+            // sync writes with is immedate == true as over Stream
```

Comment:
Is this comment explaining why we don't use `WriteAsync` here? I think we should use `WriteAsync` instead of write since the normal `SocketOutput.ProducingComplete`. We probably do need to trace async errors though.